File: src/main/scala/org/ximo/finkinaction/scala/dataset/DataSetSinkApp.scala
package org.ximo.finkinaction.scala.dataset

import org.apache.flink.api.scala.{ExecutionEnvironment, _}
import org.apache.flink.core.fs.FileSystem.WriteMode

/**
  *
  *
  * @author xikl
  * @date 2019/8/28
  */
object DataSetSinkApp extends App {

  val env = ExecutionEnvironment.getExecutionEnvironment

  val data = 1 to 10
  
  val input = env.fromCollection(data)

  // 如果是多个并发 它会是一个文件
  input.writeAsText("./test.txt", WriteMode.OVERWRITE)
      .setParallelism(3)

  // 这一步很关键
  env.execute("sink-test")


}

File: src/main/java/org/ximo/finkinaction/java/dataset/DataSetSinkApp.java
package org.ximo.finkinaction.java.dataset;

import org.apache.flink.api.java.ExecutionEnvironment;
import org.apache.flink.api.java.operators.DataSource;
import org.apache.flink.api.java.tuple.Tuple2;
import org.apache.flink.core.fs.FileSystem;
import org.ximo.finkinaction.java.constants.CommonData;

/**
 * @author xikl
 * @date 2019/8/28
 */
public class DataSetSinkApp {

    public static void main(String[] args) throws Exception {
        final ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();

        final DataSource<Tuple2<Integer, String>> text = env.fromCollection(CommonData.TUPLE2_LIST);

        text.writeAsText("./test.txt", FileSystem.WriteMode.OVERWRITE);

        env.execute("sink-test");
    }

}
